lib/resource/Directory.js

```diff
@@ -75,7 +77,19 @@ Directory.prototype.getProvider = function getProvider(/* [options,] callback */
     return callback();
   }
 
-  return this.dataStore.getResource(this.provider.href, options, require('./Provider'), callback);
+  function providerFactory(provider) {
```

Comment:
This doesn't seem to work properly, it gives me an object of type `InstanceResource`, and not `SamlProvider`:

```javascript
InstanceResource {
  href: 'http://localhost:7001/v1/directories/7MOtq5XXpQVqnD7WgpxoW4/provider',
  createdAt: '2015-12-17T21:55:54.784Z',
  modifiedAt: '2015-12-17T21:55:54.784Z',
  providerId: 'saml',
  ssoLoginUrl: 'http://login.se',
  ssoLogoutUrl: 'http://logout.se',
  encodedX509SigningCert: '-----BEGIN CERTIFICATE-----...-----END CERTIFICATE-----',
  requestSignatureAlgorithm: 'RSA-SHA1',
  attributeStatementMappingRules: {
    href: 'http://localhost:7001/v1/attributeStatementMappingRules/7MPwgswI85ZzR3RYFJ1KIm'
  },
  serviceProviderMetadata: {
    href: 'http://localhost:7001/v1/samlServiceProviderMetadatas/7NB10jttftaGM1ygGOZOei'
  }
}
```